File: articles/what-is-algorithms-computational-reasoning/java/src/main/java/org/contentcatalyst/algorithms/AlgorithmicReasoning.java
package org.contentcatalyst.algorithms;

public class AlgorithmicReasoning {
    static double clamp(double v) {
        return Math.max(0.0, Math.min(100.0, v));
    }

    public static void main(String[] args) {
        String[] names = {
            "Brute-force procedure",
            "Indexed search design",
            "Graph-aware reasoning",
            "Governed computational reasoning"
        };
        double[] representation = {40, 62, 76, 86};
        double[] correctness = {28, 52, 68, 82};
        double[] governance = {20, 38, 54, 86};
        double[] bruteForce = {92, 42, 30, 18};

        System.out.println("scenario,reasoning_score,warning");
        for (int i = 0; i < names.length; i++) {
            double score = clamp(0.30 * representation[i] + 0.30 * correctness[i] + 0.30 * governance[i] - 0.10 * bruteForce[i]);
            System.out.printf("%s,%.3f,Synthetic governance diagnostic only.%n", names[i], score);
        }
    }
}

File: articles/what-is-algorithms-computational-reasoning/prolog/algorithmic_reasoning.pl
scenario('Brute-force procedure', 40, 28, 20, 92).
scenario('Indexed search design', 62, 52, 38, 42).
scenario('Graph-aware reasoning', 76, 68, 54, 30).
scenario('Governed computational reasoning', 86, 82, 86, 18).

score(Name, Score) :-
    scenario(Name, Representation, Correctness, Governance, BruteForce),
    Raw is 0.30 * Representation + 0.30 * Correctness + 0.30 * Governance - 0.10 * BruteForce,
    Score is max(0, min(100, Raw)).
